cmd/agent/gui/render.go

```diff
@@ -38,7 +38,7 @@ func renderRunningChecks() (string, error) {
 		return "", err
 	}
 	loaderErrs := collector.GetLoaderErrors()
-	configErrs := autodiscovery.GetConfigErrors()
+	configErrs := autodiscoveryimpl.GetConfigErrors()
```

Comment:
Why are those functions part of the global autodiscovery module? Could we not use the autodiscovery component instead?

Reply:
As my reply [here](https://github.com/DataDog/datadog-agent/pull/23116#discussion_r1515095149), reasons for not putting them into component are:

1. They were not member function before the migration. Ideally we should have split the autodiscovery into comp and pkg folders, but that will make the code management difficult. 
2. We can certainly make them member functions of component. The only challenging part is to pass component into [GetPayload](https://github.com/DataDog/datadog-agent/blob/dd8188170eb12b0964a29426cf42fc8bd9e9c69b/pkg/collector/metadata/agentchecks/agentchecks.go#L25) where we usually called pkg global function like collector.GetLoaderErrors()

Reply:
Hey @zhuminyi 

Thanks for the context. 

It is a shame that the `GetPayload` can not easily get access to the autodiscovery component. Looking at the code for the expvar errors for autodiscovery. It seems that we still relay in globals as you mention. Would be good to consider removing those global variables (Work for another PR) Also, it would be good to contact @gh123man regarding passing the autodiscory component to the collector. He recently worked on componetizing the collector

Reply:
Getting the AD component into `GetPayload` doesn't look like that much work. The call hierarchy is as follows: 

- `GetPayload` is [called by `AgentChecksCollector.Send`](https://github.com/DataDog/datadog-agent/blob/5d68596816649140d4305aacf430dd803cd7748d/pkg/collector/metadata/agentchecks.go#L23)
- `Send` [is called by the metadata scheduler](https://github.com/DataDog/datadog-agent/blob/5d68596816649140d4305aacf430dd803cd7748d/pkg/metadata/scheduler.go#L63) which already has references to components (demux) 

So you could add the ad component as a dependency to the [metadata scheduler](https://github.com/DataDog/datadog-agent/blob/5d68596816649140d4305aacf430dd803cd7748d/pkg/metadata/scheduler.go#L42) which gets created at the agent entrypoints where all components are available. 

Not sure if this should be in a followup PR - but it looks like a fairly small amount of effort to thread the component through.